### DISCUSSION TITLE
Load dynamic rows depending on Master table value

### DISCUSSION THREAD

**Discussion by rhemasystems** [serenity-is/Serenity#6632] [Help] [answered]
I have 3 tables which are Master table,Details table  and DetailsHistory tables
Currently when I click on a row in the Master table grid on Serenity it loads the Details table when the dialog form opens which is perfect as shown in the below image.
What I want is for Serenity to load the data from either the Details or the DetailsHistory table depending on a field value in the Master table.
Can anyone assist.

![image](https://user-images.githubusercontent.com/50826372/209806700-02e0238d-5260-4b45-91ac-32e063dd035d.png)

**Comment by wezmag** [comment, marked as answer]
You can put both Detail and DetailHistory on the form.
When the field changed, change the visibility of the Detail or the DetailHistory grid.

**Reply by rhemasystems** [reply]
Thank you, but please can you assist me on how to hide or show the grid

**Reply by wezmag** [reply]
Use following code to show or hide the field in a form.
``` ts
//to show
this.form.FieldName.element.closest('.field').show();
this.form.FieldName.element.closest('.field').toggle(true);
//to hide
this.form.FieldName.element.closest('.field').hide();
this.form.FieldName.element.closest('.field').toggle(false);
```

Assume the field in your form is called `Type`, and it's a dropdown with two values, `present` and `history`.

In your dialog.ts, you want to bind an event to the field in the constructor, that event will toggle visibility of the Details and DeatilHistory.
``` ts
this.form.Type.changeSelect2((e) => {
    this.form.Details.element.closest('.field').toggle(this.form.Type.value == 'present');
    this.form.DetailHistory.element.closest('.field').toggle(this.form.Type.value == 'history');
});
```
And then you may also want to set initial value for the dropdown and the visibility of Details and DetailHistory for both New and EditMode
``` ts
protected updateInterface() {
    super.updateInterface();
    if (this.isNew())
    {
        this.form.Type.value = 'present';
        this.form.Details.element.closest('.field').toggle(this.form.Type.value == 'present');
        this.form.DetailHistory.element.closest('.field').toggle(this.form.Type.value == 'history');
    }

    if (this.isEditMode())
    {
        this.form.Details.element.closest('.field').toggle(this.form.Type.value == 'present');
        this.form.DetailHistory.element.closest('.field').toggle(this.form.Type.value == 'history');
    }
}
```
